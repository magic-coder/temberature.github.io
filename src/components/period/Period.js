import React from 'react'

export default class Period extends React.Component {
    render () {
        const { course, images } = this.props;
        if (!course||!images) {
            return null;
        }
        const now = new Date().getTime();
        if (now < course.event_register_deadline) {
          return <img src={images[0]} alt="" />;
        } else if (now > course.event_start_date && now < course.event_end_date) {
          return <img src={images[1]} alt="" />;
        } else if (now > course.event_end_date) {
          return <img src={images[2]} alt="" />;
        } else {
            return <img alt="" />
        }
    }
}