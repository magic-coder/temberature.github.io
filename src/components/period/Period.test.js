// import { configure } from "enzyme";
// import Adapter from "enzyme-adapter-react-15";
// configure({ adapter: new Adapter() });
import React from "react";
// import ReactDOM from "react-dom";
import Period from "./Period";
// import { shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";

const now = Date.now();
const weekTime = 7 * 24 * 60 * 60;

it("valid data", () => {
  const cases = new Map([
    [
      "enrolling",
      {
        event_register_deadline: now + weekTime,
        event_start_date: now + 2 * weekTime,
        event_end_date: now + 3 * weekTime,
      }
    ],
    [
      "ongoing",
      {
        event_register_deadline: now - 2 * weekTime,
        event_start_date: now - weekTime,
        event_end_date: now + weekTime,
      }
    ],
    [
      "finish",
      {
        event_register_deadline: now - 3 * weekTime,
        event_start_date: now - 2 * weekTime,
        event_end_date: now - weekTime,
      }
    ]
  ]);


  cases.forEach((value, key) => {
    const tree = renderer
    .create(
      <Period
        course={value}
        images={[
          "period_enrolling.png",
          "period_ongoing.png",
          "period_finish.png"
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});

it("invalid data", () => {
  const cases = new Map([
    [
      "invalid",
      {
        event_register_deadline: now + 3 * weekTime,
        event_start_date: now - weekTime,
        event_end_date: now - 2 * weekTime,
      }
    ]
  ]);

  cases.forEach((value, key) => {
    const tree = renderer
    .create(
      <Period
        course={value}
        images={[
          "period_enrolling.png",
          "period_ongoing.png",
          "period_finish.png"
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});
