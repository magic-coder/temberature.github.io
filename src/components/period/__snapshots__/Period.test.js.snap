// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid data 1`] = `
<img
  alt=""
  src=""
/>
`;

exports[`valid data 1`] = `
<img
  alt=""
  src="period_enrolling.png"
  style={
    Object {
      "verticalAlign": "bottom",
    }
  }
/>
`;

exports[`valid data 2`] = `
<img
  alt=""
  src="period_ongoing.png"
  style={
    Object {
      "verticalAlign": "bottom",
    }
  }
/>
`;

exports[`valid data 3`] = `
<img
  alt=""
  src="period_finish.png"
  style={
    Object {
      "verticalAlign": "bottom",
    }
  }
/>
`;
